ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayLayerViewDrawStatus/DisplayLayerViewDrawStatus.h

```diff
@@ -49,25 +47,46 @@ class DisplayLayerViewDrawStatus : public QQuickItem
   static void init();
 
 signals:
-  void statusChanged();
-  void namesChanged();
-  void mapReady();
+  void modelChanged();
 
 private:
-  QStringList layerViewStates() const;
-  QStringList layerNames() const;
+  const QList<QObject*>& statusModel() const;
   void addLayers();
   void connectSignals();
 
 private:
+  QList<QObject*> m_statuses;
   Esri::ArcGISRuntime::Map* m_map = nullptr;
   Esri::ArcGISRuntime::MapQuickView* m_mapView = nullptr;
   Esri::ArcGISRuntime::ArcGISMapImageLayer* m_imageLayer = nullptr;
   Esri::ArcGISRuntime::ArcGISTiledLayer* m_tiledLayer = nullptr;
   Esri::ArcGISRuntime::ServiceFeatureTable* m_featureTable = nullptr;
   Esri::ArcGISRuntime::FeatureLayer* m_featureLayer = nullptr;
-  QStringList m_layerNames;
-  QStringList m_layerViewStates;
+};
+
+class DisplayItem : public QObject
+{
+  Q_OBJECT
+  Q_PROPERTY(QString name READ name WRITE setName NOTIFY nameChanged)
+  Q_PROPERTY(QString status READ status WRITE setStatus NOTIFY statusChanged)
+public:
+  DisplayItem(const QString& name, const QString& status, QObject* parent = nullptr);
+
+  using QObject::QObject;
+
+  void setName(const QString& name);
+  const QString& name() const;
+
+  void setStatus(const QString& status);
+  const QString& status() const;
```

Comment:
This and above should return values. QString is reference counted so cheap to copy. Not much savings here and could get into trouble if the values change unexpectedly.

Reply:
That's fair. Will rely on the COW.